src/analysis/retail/priest/shadow/modules/talents/InescapableTorment.tsx

```diff
@@ -21,13 +29,74 @@ class InescapableTorment extends Analyzer {
       Events.damage.by(SELECTED_PLAYER_PET).spell(SPELLS.INESCAPABLE_TORMENT_TALENT_DAMAGE),
       this.onDamage,
     );
+    this.addEventListener(
+      Events.cast.by(SELECTED_PLAYER).spell(TALENTS.MINDBENDER_SHADOW_TALENT),
+      this.onCast,
+    );
+    this.addEventListener(Events.fightend, this.onEnd);
+  }
+
+  onCast(event: CastEvent) {
+    //Since there is no way to tell when mindbender ends, we resolve the previous Mindbender when a new one is cast.
+    //If this is not the first mindbender cast, then we generate the extension.
+    if (this.castTime !== 0) {
+      const tooltip = (
+        <>
+          @<strong>{this.owner.formatTimestamp(this.castTime)}</strong>, Extension:
+          <strong>{this.extension.toFixed(1)}</strong>
+        </>
+      );
+
+      let value = QualitativePerformance.Good;
+      if (this.extension <= 6) {
+        value = QualitativePerformance.Ok;
+      }
+      if (this.extension <= 3) {
+        value = QualitativePerformance.Fail;
+      }
+
+      this.MBExtension.push({ value, tooltip });
+    }
+
+    this.totalTime += this.extension; // add previous extension to total time.
+    this.castTime = event.timestamp; //get cast time for current mindbender.
+    this.extension = 0; // reset extension.
+  }
+
+  onEnd(event: FightEndEvent) {
+    //Since there is no way to tell when mindbender ends, we resolve the last Mindbender when the fight is over.
+    //So long as a mindbender was cast, we generate the extension.
+    if (this.castTime !== 0) {
+      const tooltip = (
+        <>
+          @<strong>{this.owner.formatTimestamp(this.castTime)}</strong>, Extension:
+          <strong>{this.extension.toFixed(1)}</strong>
+        </>
+      );
+
+      let value = QualitativePerformance.Good;
+      if (this.extension <= 6) {
+        value = QualitativePerformance.Ok;
+      }
+      if (this.extension <= 3) {
+        value = QualitativePerformance.Fail;
+      }
+
+      this.MBExtension.push({ value, tooltip });
+    }
+
+    this.totalTime += this.extension; // add previous extension to total time.
```

Comment:
instead of a copy/paste between this and `onCast`, you could put them in a shared method (e.g. `protected finalizeMindBenderCast`). `onCast` needs a little more because it then sets up a new mind bender cast, but they're mostly shared.

Reply:
I've changed it to use a protected method.